src/Forcings/relaxation.jl

```diff
@@ -154,6 +157,63 @@ show_exp_arg(D, c) = c == 0 ? "$D^2" :
 Base.summary(g::GaussianMask{D}) where D =
     "exp(-$(show_exp_arg(D, g.center)) / (2 * $(g.width)^2))"
 
+
+"""
+    PiecewiseLinearMask{D}(center, width)
+
+Callable object that returns a piecewise linear masking function centered on
+`center`, with `width`, and varying along direction `D`. The mask is:
+- 0 when |D - center| > width
+- 1 when D = center
+- Linear interpolation between 0 and 1 when |D - center| ≤ width
+
+Example
+=======
+
+Create a piecewise linear mask centered on `z=0` with width `1` meter.
+
+```jldoctest
+julia> using Oceananigans
+
+julia> mask = PiecewiseLinearMask{:z}(center=0, width=1)
+PiecewiseLinearMask{:z, Int64}(0, 1)
+
+julia> mask(0, 0, 0) == 1
+true
+
+julia> mask(0, 0, 1) == mask(0, 0, -1) == 0
+true
+```
+"""
+struct PiecewiseLinearMask{D, T}
+    center :: T
+     width :: T
+
+    function PiecewiseLinearMask{D}(; center, width) where D
+        T = promote_type(typeof(center), typeof(width))
+        return new{D, T}(center, width)
+    end
+end
+
+@inline function (p::PiecewiseLinearMask{:x})(x, y, z)
+    d = abs(x - p.center)
+    return d > p.width ? 0.0 : 1.0 - d/p.width
+end
```

Comment:
```suggestion
    d = 1 - abs(x - p.center) / p.width
    return max(zero(d), d)
end
```

There are two issues with the code as written:

1. It hardcodes Float64 in `0.0` and `1.0`
2. It invokes short-circuiting logic but doesn't need to

Reply:
Ah, yeah, thanks for catching that. Done.